tianshou/data/collector.py

```diff
@@ -177,43 +172,44 @@ def reset_buffer(self, keep_statistics: bool = False) -> None:
         """Reset the data buffer."""
         self.buffer.reset(keep_statistics=keep_statistics)
 
-    def reset_env(self, gym_reset_kwargs: dict[str, Any] | None = None) -> None:
+    def reset_env(
+        self,
+        gym_reset_kwargs: dict[str, Any] | None = None,
+    ) -> tuple[np.ndarray, dict | list[dict]]:
         """Reset all of the environments."""
         gym_reset_kwargs = gym_reset_kwargs if gym_reset_kwargs else {}
         obs, info = self.env.reset(**gym_reset_kwargs)
-        if self.preprocess_fn:
-            processed_data = self.preprocess_fn(obs=obs, info=info, env_id=np.arange(self.env_num))
-            obs = processed_data.get("obs", obs)
-            info = processed_data.get("info", info)
-        self.data.info = info  # type: ignore
-        self.data.obs = obs
-
-    def _reset_state(self, id: int | list[int]) -> None:
-        """Reset the hidden state: self.data.state[id]."""
-        if hasattr(self.data.policy, "hidden_state"):
-            state = self.data.policy.hidden_state  # it is a reference
+        return obs, info
+
+    def _reset_state(
+        self,
+        cur_rollout_batch: RolloutBatchProtocol,
+        id: int | list[int],
+    ) -> RolloutBatchProtocol:
+        """Reset the hidden state: cur_rollout_batch.state[id]."""
+        if hasattr(cur_rollout_batch.policy, "hidden_state"):
+            state = cur_rollout_batch.policy.hidden_state  # it is a reference
             if isinstance(state, torch.Tensor):
                 state[id].zero_()
             elif isinstance(state, np.ndarray):
                 state[id] = None if state.dtype == object else 0
             elif isinstance(state, Batch):
                 state.empty_(id)
+        return cur_rollout_batch
 
     def _reset_env_with_ids(
         self,
+        cur_rollout_batch: RolloutBatchProtocol,
         local_ids: list[int] | np.ndarray,
         global_ids: list[int] | np.ndarray,
         gym_reset_kwargs: dict[str, Any] | None = None,
-    ) -> None:
+    ) -> RolloutBatchProtocol:
```

Comment:
Rename `cur_rollout_batch` to `rollout_batch`

The method modifies the batch, it shouldn't return it. The name should also be adjusted, I'm generally not sure the method should exist (is it used in many places?)